src/app/system/account-number-preferences/edit-account-number-preference/edit-account-number-preference.component.ts

```diff
@@ -0,0 +1,88 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Services */
+import { SystemService } from 'app/system/system.service';
+
+/**
+ * Edit Account Number Preference Component.
+ */
+@Component({
+  selector: 'mifosx-edit-account-number-preference',
+  templateUrl: './edit-account-number-preference.component.html',
+  styleUrls: ['./edit-account-number-preference.component.scss']
+})
+export class EditAccountNumberPreferenceComponent implements OnInit {
+
+  /** Account Number Preference Form */
+  accountNumberPreferenceForm: FormGroup;
+  /** Account Number Preference Data */
+  accountNumberPreferenceData: any;
+  /** Account Number Preferences Template Data */
+  accountNumberPreferencesTemplateData: any;
+  /** Prefix Type Data */
+  prefixTypeData: any[];
+
+  /**
+   * Retrieves the account number preference and account number preferences template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService Accounting Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private route: ActivatedRoute,
+              private formBuilder: FormBuilder,
+              private systemService: SystemService,
+              private router: Router) {
+    this.route.data.subscribe((data: { accountNumberPreference: any, accountNumberPreferencesTemplate: any }) => {
+      this.accountNumberPreferenceData = data.accountNumberPreference;
+      this.accountNumberPreferencesTemplateData = data.accountNumberPreferencesTemplate;
+    });
+  }
+
+  /**
+   * Creates and sets account number preference form.
+   */
+  ngOnInit() {
+    this.createAccountNumberPreferenceForm();
+  }
+
+  /**
+   * Sets Prefix type data.
+   * Creates and sets the edit account number preference form.
+   */
+  createAccountNumberPreferenceForm() {
+    let prefixTypeId = '';
+    this.prefixTypeData = this.accountNumberPreferencesTemplateData.prefixTypeOptions[this.accountNumberPreferenceData.accountType.code];
+    if (this.accountNumberPreferenceData['prefixType'] !== undefined) {
+      for (const prefixTypeOption of this.prefixTypeData) {
```

Comment:
You don't need to use a for loop here. Just directly assign `this.accountNumberPreferenceData.prefixType.id` while building the form.